### PR TITLE
T19: Factorial of a Number

### PR DESCRIPTION
**Pull request by Ashish02112001**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+   /// <summary>Factorial of a number</summary>
    internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>This method asks the number from the user and gives factorial of it</summary>
+      /// <param name="args">arguments</param>
       static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+         Console.Write ("Enter a Number: ");
+         for(; ; ) {
+            if (int.TryParse (Console.ReadLine (), out var num) && num > 0 && num <= 20) {
```
Comment (asfar-nazar) [inline Program.cs:18]: Try and avoid the limit at 20!. Hint: search for System.Numeric

**Review by asfar-nazar — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 1 above)
Comment (Ashish02112001) [reply Program.cs:18]: Are you telling to use the byte na?

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+      #region Methods ---------------------------------------------
+      /// <summary>This method asks the number from the user and gives factorial of it</summary>
+      /// <param name="args">arguments</param>
       static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+         Console.Write ("Enter a Number: ");
+         for (; ; ) {
+            if (int.TryParse (Console.ReadLine (), out var num) && num > 0) {
+               Console.WriteLine ($"{num}! = {Factorial (num)}"); break;
+            } else Console.Write ("Enter a valid number below 20: ");
```
Comment (asfar-nazar) [inline Program.cs:21]: Is this else block necessary?

**Review by asfar-nazar — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+            if (int.TryParse (Console.ReadLine (), out var num) && num > 0) {
+               Console.WriteLine ($"{num}! = {Factorial (num)}"); break;
+            } 
+         }
       }
+      #endregion
+      /// <summary>Gets the number and calculates the factorial of it</summary>
+      /// <param name="num">Number from the user</param>
+      /// <returns>Factorial of the number</returns>
+      static BigInteger Factorial (int num) => (num > 1) ? num * Factorial (num - 1) : 1;
```
Comment (asfar-nazar) [inline Program.cs:28]: Will this work if input is 10000? Please check.

**Review by asfar-nazar — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 3 above)
Comment (Ashish02112001) [reply Program.cs:28]: Not Working na It's Crashing.

**Review by asfar-nazar — APPROVED** [review]

**Pull request closed without merging**